volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -236,53 +264,53 @@ def parse_data_record(
             content = renderers.NotAvailableValue()
 
         yield (
-            format_hints.Hex(record_map[mft_record.vol.offset][2]),
-            mft_record.get_signature(),
+            format_hints.Hex(record_map[mft_record.vol.offset].offset),
+            str(mft_record.get_signature()),
             mft_record.RecordNumber,
             attr.Attr_Header.AttrType.lookup(),
-            record_map[mft_record.vol.offset][0],
+            record_map[mft_record.vol.offset].record_name
+            or renderers.NotAvailableValue(),
             ads_name,
             content,
         )
 
     @classmethod
     def parse_data_records(
         cls,
-        record_map: Dict[int, Tuple[str, int, int]],
-        mft_record: interfaces.objects.ObjectInterface,
-        attr: interfaces.objects.ObjectInterface,
+        record_map: DefaultDict[str, MFTRecord],
+        mft_record: mft.MFTEntry,
+        attr: mft.MFTAttribute,
         symbol_table_name: str,
         return_first_record: bool,
-    ) -> Generator[Iterable, None, None]:
+    ) -> Iterator[Tuple]:
         """
         Parses DATA records while maintaining the FILE_NAME association
         from previous parsing of the record
         Suports returning the first/main $DATA as well as however many
         ADS records a file might have
         """
-        if mft_record.vol.offset not in record_map:
-            # file name, DATA count, offset
-            record_map[mft_record.vol.offset] = [renderers.NotAvailableValue(), 0, None]
+        rec = record_map[mft_record.vol.offset]
```

Comment:
Oh, now I see why you used the `DefaultDict`.  Given that `namedtuples` need to be constructed initially anyway (and then can't be changed), I'd sooner build up the component, construct the `namedtuple` and then assign it into the record_map.  I'm guessing this doesn't ever overwrite existing answers (otherwise that's also a conflict we should cope with directly, rather than just blatting over prior results without noting whether there's a discrepancy)...